src/main/java/com/faforever/client/game/GenerateMapController.java

```diff
@@ -358,6 +373,20 @@ protected void setStyles(List<String> styles) {
     mapStyleLabel.setVisible(true);
   }
 
+  protected void setBiomes(List<String> biomes) {
+    biomes.add(0, MapGeneratorService.GENERATOR_RANDOM_BIOME);
```

Comment:
Looking at this now this and set styles really should make a copy of the list that is guaranteed to be mutable

Reply:
how do I do that and why is that desirable?

Reply:
It is desirable because this method could be passed an immutable list so the add function would fail.

Or if a mutable list is passed this then modifies that list in ways that the caller may not be expecting so it breaks something higher up.

You would implement it by just creating a new list and then adding this list to that one.